adit/core/processors.py

```diff
@@ -131,11 +135,27 @@ def _transfer_to_archive(self) -> None:
             patient_folder = self._download_to_folder(Path(tmpdir))
             _add_to_archive(archive_path, archive_password, patient_folder)
 
-    def _transfer_to_folder(self) -> None:
-        assert self.transfer_task.destination.node_type == DicomNode.NodeType.FOLDER
-        dicom_folder = Path(self.transfer_task.destination.dicomfolder.path)
-        download_folder = dicom_folder / self._create_destination_name()
+    def _convert_dicom_to_nifti(self, dicom_folder: Path) -> None:
+        nifti_folder = (
+            Path(self.transfer_task.destination.dicomfolder.path) / self._create_destination_name()
+        )
+
+        converter = DicomToNiftiConverter()
+        converter.convert(dicom_folder, nifti_folder)
+
+    def _get_download_folder(self, use_temp_folder: bool) -> Path:
+        if use_temp_folder:
+            with tempfile.TemporaryDirectory(prefix="adit_") as tmpdir:
+                return Path(tmpdir) / self._create_destination_name()
```

Comment:
This won't work. Read about how context managers work in Python in general. The cleanup happens as soon as the context manager is left.